### DISCUSSION TITLE
modules and encapsulation

### DISCUSSION THREAD

**Discussion by RupertAvery** [Maximilian-Winter/Bite-Programming-Language#43] [General]
I think modules should have encapsulation.

When you `import` and `using` a module, you allow name resolution to bubble up to the imported module.

If the module has many variables, this may cause collisions in the calling module.  The more modules you import, the more it becomes a problem.

classes and functions are less of a problem than variables.  And sometimes you don't want to expose module variables to importers, because you are just using them in your module.

I suggest we add an `public` modifier at the module level to control what methods, classes and variables are accessible from other modules. If not specified, any declaration should be private.

In TypeScript modules, the equivalent is `export`.

In Go, encapsulation is controlled by the case of the first character of a type, class or function.  lowerFirstCase = private, UpperFirstCase = public

## Example

```
module FunStuff;

class internal() {
}

public class foo() {
    internal():
}
```

```
module Main;

import FunStuff;

FunStuff.foo(); // succeeds

FunStuff.internal();  // fails, no exported declaration 'internal'

```

**Comment by Maximilian-Winter** [comment]
I'm already planning to implement access modifiers, also for module level
